src/App.tsx

```diff
@@ -15,36 +16,85 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABETICALLY = 'sort-alphabetically',
+  BY_LENGTH = 'sort-by-length',
+  NONE = '',
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  let visibleGoods = [...goodsFromServer].sort((good1, good2) => {
+    switch (sortField) {
+      case SortType.ALPHABETICALLY:
+        return good1.localeCompare(good2);
+      case SortType.BY_LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
```

Comment:
[CODE KNOWLEDGE] - You're interacting with the DOM directly by sorting the goods array every time the component re-renders. This can lead to performance issues. It would be better to use the `useEffect` hook to sort the array only when `sortField` or `reversed` changes.